server/src/ZetaSqlWrapper.ts

```diff
@@ -131,6 +137,30 @@ export class ZetaSqlWrapper {
     }
   }
 
+  async getTempUdfs(modelFetcher: ModelFetcher): Promise<FunctionProto[]> {
+    const model = await modelFetcher.getModel();
+    if (model?.config?.sqlHeader) {
+      const languageOptions = await this.getLanguageOptions();
+      return this.sqlHeaderAnalyzer.getAllFunctionDeclarations(
+        model.config.sqlHeader,
+        languageOptions?.serialize(),
+        this.catalog.builtinFunctionOptions,
+      );
+    }
+    return [];
+  }
+
+  addTempUdfsToCatalog(catalog: SimpleCatalogProto, udfs: FunctionProto[]): SimpleCatalogProto {
+    const clonedCatalog = { ...catalog };
```

Comment:
Yes. The idea is to have `current catalog` (`this.catalog`) + `temp functions` for the current model. If we add temp function to `this.catalog` we will need delete it after analyzing because temp functions works only for current model.